src/System.Data.Common/src/System/Data/Common/DbConnectionOptions.cs

```diff
@@ -464,17 +466,18 @@ private static Hashtable SplitConnectionString(string connectionString, Hashtabl
             return parsetable;
         }
 
-        private static void ParseComparison(Hashtable parsetable, string connectionString, Hashtable synonyms, bool firstKey, Exception e)
+        private static void ParseComparison(Dictionary<string, string> parsetable, string connectionString, Dictionary<string, string> synonyms, bool firstKey, Exception e)
         {
             try
             {
-                Hashtable parsedvalues = SplitConnectionString(connectionString, synonyms, firstKey);
-                foreach (DictionaryEntry entry in parsedvalues)
+                Dictionary<string, string> parsedvalues = SplitConnectionString(connectionString, synonyms, firstKey);
+                foreach (KeyValuePair<string, string> entry in parsedvalues)
                 {
-                    string keyname = (string)entry.Key;
-                    string value1 = (string)entry.Value;
-                    string value2 = (string)parsetable[keyname];
-                    Debug.Assert(parsetable.Contains(keyname), "ParseInternal code vs. regex mismatch keyname <" + keyname + ">");
+                    string keyname = entry.Key;
+                    string value1 = entry.Value;
+                    string value2 = null;
+                    parsetable.TryGetValue(keyname, out value2);
```

Comment:
Why are you switching to TryGetValue? Does the index throw in dictionary but not in hashtable?